python/ray/experimental/data/tests/test_dataset.py

```diff
@@ -104,6 +105,90 @@ def test_pyarrow(ray_start_regular_shared):
+    assert 190 == sum([dataset.sum() for dataset in datasets])
+
+
+def test_split_hints(ray_start_regular_shared):
+    @ray.remote
+    class Actor(object):
+        def __init__(self):
+            pass
+
+    def test_helper(block_node_ids, actor_node_ids, expected_split_result):
+        """Helper function to setup split hints test.
+
+        Args:
+            block_node_ids: a list of blocks with their locations. For
+                example ["node1", "node2"] represents two blocks with
+                "node1", "node2" as their location respectively.
+            actor_node_ids: a list of actors with their locations. For
+                example ["node1", "node2"] represents two actors with
+                "node1", "node2" as their location respectively.
+            expected_split_result: a list of allocation result, each entry
+                in the list stores the block_index in the split dataset.
+                For example, [[0, 1], [2]] represents the split result has
+                two datasets, datasets[0] have blocks 0 and 1; and 
+                datasets[1] have blocks 2.
+        """
+        num_blocks = len(block_node_ids)
+        ds = ray.experimental.data.range(num_blocks, parallelism=num_blocks)
+        blocks = list(ds._blocks)
+        assert len(block_node_ids) == len(blocks)
+        actors = [Actor.remote() for i in range(len(actor_node_ids))]
+        with patch("ray.experimental.get_object_locations") as location_mock:
+            with patch("ray.state.actors") as state_mock:
+                block_locations = {}
+                for i, node_id in enumerate(block_node_ids):
+                    if node_id:
+                        block_locations[blocks[i]] = {"node_ids": [node_id]}
+                location_mock.return_value = block_locations
+
+                actor_state = {}
+                for i, node_id in enumerate(actor_node_ids):
+                    actor_state[actors[i]._actor_id.hex()] = {
+                        "Address": {
+                            "NodeID": node_id
+                        }
+                    }
+
+                state_mock.return_value = actor_state
+
+                datasets = ds.split(len(actors), actors)
+                assert len(datasets) == len(actors)
+                for i in range(len(actors)):
+                    assert set(
+                        [blocks[j] for j in expected_split_result[i]]) == set(
+                            datasets[i]._blocks)
+
+    test_helper(["node2", "node1", "node1"], ["node1", "node2"], [[1, 2], [0]])
+    test_helper(["node1", "node1", "node1"], ["node1", "node2"], [[0, 1], [2]])
+    test_helper(["node2", "node2", "node2"], ["node1", "node2"], [[1, 2], [0]])
+    test_helper(["node2", "node2", None], ["node1", "node2"], [[1, 2], [0]])
+    test_helper(["node2", "node2", None], [None, None], [[2, 0], [1]])
+    test_helper(["n1", "n2", "n3", "n1", "n2"], ["n1", "n2"],
```

Comment:
Could you add test cases with a much larger number of blocks (e.g., 100 per node?)